File: week01/완전탐색/10819.py
from itertools import permutations
# N = 8, 8개의 배열을 나열하는 방법은 8!
N = int(input())

num_list = list(map(int, input().split()))

def get_sum(N, arr):
  result = 0
  for i in range(1, N):
    result += abs(arr[i] - arr[i - 1])
  return result

# 그럼 N개의 원소를 가진 배열을 싹 나열후 각각 배열의 합을 구하면?

temp = list(map(list, permutations(num_list, N)))
sum_list = []

for cases in temp:
  total = get_sum(N, cases)
  sum_list.append(total)

print(max(sum_list))

File: week01_JS/완전탐색/10819.js
// 배열의 순서를 적절히 변경하여 수식의 결과값을 최대로

const fs = require('fs')
const input = fs.readFileSync('input.txt').toString().trim().split('\n');

// 순열을 구현하고 각각 순열의 합 중 최대 합을 리턴

let N = parseInt(input[0])
let num_list = input[1].split(' ').map(Number)


const getPerm = (arr, n) => {
  if (n == 1) return arr.map(el => [el]);
  const result = [];

  arr.forEach((fixed, idx, origin) => {
    const rest = [...origin.slice(0, idx), ...origin.slice(idx + 1)]
    const perms = getPerm(rest, n - 1)
    const attached = perms.map(perm => [fixed, ...perm])
    result.push(...attached)
  });
  return result
}

perm_list = getPerm(num_list, N)
result_list = []

for (let perm of perm_list) {
  let result = 0
  for (let i = 1; i < perm.length; i++) {

    result += Math.abs(perm[i] - perm[i - 1])
  }
  result_list.push(result)
}

console.log(Math.max(...result_list))